precompiles/staking/staking.go

```diff
@@ -74,21 +63,42 @@ func initABI() {
 type Contract struct {
 	ptypes.BaseContract
 
-	stakingKeeper stakingkeeper.Keeper
-	cdc           codec.Codec
-	kvGasConfig   storetypes.GasConfig
+	stakingKeeper  stakingkeeper.Keeper
+	fungibleKeeper fungiblekeeper.Keeper
+	bankKeeper     bankkeeper.Keeper
+	zrc20ABI       *abi.ABI
+	cdc            codec.Codec
+	kvGasConfig    storetypes.GasConfig
 }
 
 func NewIStakingContract(
+	ctx sdk.Context,
 	stakingKeeper *stakingkeeper.Keeper,
+	fungibleKeeper fungiblekeeper.Keeper,
+	bankKeeper bankkeeper.Keeper,
 	cdc codec.Codec,
 	kvGasConfig storetypes.GasConfig,
 ) *Contract {
+	accAddress := sdk.AccAddress(ContractAddress.Bytes())
+	if fungibleKeeper.GetAuthKeeper().GetAccount(ctx, accAddress) == nil {
+		fungibleKeeper.GetAuthKeeper().SetAccount(ctx, authtypes.NewBaseAccount(accAddress, nil, 0, 0))
+	}
+
+	// Instantiate the ZRC20 ABI only one time.
+	// This avoids instantiating it every time deposit or withdraw are called.
+	zrc20ABI, err := zrc20.ZRC20MetaData.GetAbi()
+	if err != nil {
+		return nil
```

Comment:
Can we log the error here?

Reply:
Applied in https://github.com/zeta-chain/node/commit/06d0c5d5a8a2d7ce54f84a666c7bef7784963ccf